Should add documentation for additional classes

duergner opened the pull request:
Will add JavaDoc for all public non overridden methods on AccessTokenConfiguration and AccessTokensBuilder

See #50

@jbellmann / @harti2006 please have a look and let me know whether I got the functionality correct.

Is there a specific reason for `getGrantType()` being `public` compared to `getTokenId()` and `getScopes()` on `AccessTokensConfiguration`?

duergner commented:
Furthermore - I'd suggest to leave that PR open until the essential classes have been documented.

ePaul commented on src/main/java/org/zalando/stups/tokens/AccessTokenConfiguration.java:
```diff
@@ -26,6 +26,19 @@
 import org.zalando.stups.tokens.util.Objects;
 
+/**
+ * Configuration for one single access token that is manages by the {@link AccessTokens} service.
```
manages → managed

jbellmann commented on src/main/java/org/zalando/stups/tokens/AccessToken.java:
```diff
@@ -37,26 +49,63 @@ protected AccessToken(final String token, final String type, final long initialV
     }
 
+    /**
+     * The point in time until when this {@link AccessToken} is valud
```
should be 'is valid' instead of 'is valud'

duergner replied:
@jbellmann  Fixed on the latest push

jbellmann commented on src/main/java/org/zalando/stups/tokens/AccessToken.java:
```diff
@@ -37,26 +49,63 @@ protected AccessToken(final String token, final String type, final long initialV
+    /**
+     * Tell whether this {@link AccessToken} is expired.
+     *
+     * @return  Will return {@link Boolean#TRUE} if this {@link AccessToken} is still valid, i.e.
```
I'm not sure about this. If the AccessToken is expired I would expect an TRUE as return value. If the AccessToken is still valid, I would expect an FALSE as return value, right?

duergner replied:
You're absolutely right - will change it

duergner replied:
@jbellmann fixed on the latest push

jbellmann commented on src/main/java/org/zalando/stups/tokens/ClientCredentials.java:
```diff
@@ -15,7 +15,28 @@
 public interface ClientCredentials {
+
+    /**
+     * Get the <i>id</i> for this {@link ClientCredentials}; this is called the <i>client_id</i> on
```
Would it make sense to rename them according the OAuth2.0 spec?

duergner replied:
Do you mean the method itself or simply the description of it?

jbellmann replied:
The method. Would it better to be renamed to 'getClientId()'?

duergner replied:
But that be an API level change. Should involve a deprecation period and once renamed a major version bump, right?

jbellmann commented on src/main/java/org/zalando/stups/tokens/ClientCredentials.java:
```diff
@@ -15,7 +15,28 @@
+
+    /**
+     * Get the <i>secret</i> for this {@link ClientCredentials}; this is called the
+     * <i>client_secret</i> on the OAuth2.0 spec.
```
Same here.

jbellmann replied:
Maybe 'getClientSecret()'

duergner replied:
See comment from above

duergner commented:
@jbellmann I addressed your review comments - I wouldn't change the method names to match the OAuth spec with this PR as it's only about documentation.

But agree that it might make sense to rename them (i.e. deprecate the current ones, add the new ones) but I think this should lead to a major version bump as it's an API level breaking change in my opinion.

duergner commented:
If there are no further comments I'd suggest to merge this PR and I'd create a new one for any further documentation to not have the PR open for even longer

jbellmann commented:
👍

harti2006 commented:
:+1:
